v1.1/create-security-certificates-openssl.md

```diff
@@ -0,0 +1,292 @@
+
+## Configuration Files
+
+To use `openssl req` and `openssl ca` subcommands, you need the following configuration files:
+
+File name pattern | File usage
+-------------|------------
+`ca.conf`     | CA configuration file
+`node.conf`   | Server configuration file
+`client.conf` | Client configuration file
+
+## Examples
+
+### Generate the CA key and certificate pair
+
+1. Create two directories:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ mkdir certs
+    ~~~
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ mkdir my-safe-directory
+    ~~~
+    - `certs`: You'll generate your CA certificate and all node and client certificates and keys in this directory and then upload some of the files to your nodes.
+    - `my-safe-directory`: You'll generate your CA key in this directory and then reference the key when generating node and client certificates. After that, you'll keep the key safe and secret; you will not upload it to your nodes.
+
+2. Create the ca.conf file
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    [ ca ]
+    default_ca = CA_default
+
+    [ CA_default ]
+    default_days = 3660
+    database = index.txt
+    serial = serial.txt 
+    default_md = sha256 
+    copy_extensions = copy
+
+    # Used to generate the CA certificate.
+    [ req ]
+    prompt=no
+    distinguished_name = distinguished_name
+    x509_extensions = extensions
+
+    [ distinguished_name ]
+    organizationName = Cockroach
+    commonName = Cockroach CA
+
+    [ extensions ]
+    keyUsage = critical,digitalSignature,nonRepudiation,keyEncipherment,keyCertSign
+    basicConstraints = critical,CA:true,pathlen:1
+
+    # Common policy for nodes and users.
+    [ signing_policy ]
+    organizationName = supplied
+    commonName = supplied
+
+    # Used to sign node certificates.
+    [ signing_node_req ]
+    keyUsage = critical,digitalSignature,keyEncipherment
+    extendedKeyUsage = serverAuth,clientAuth
+
+    # Used to sign client certificates.
+    [ signing_client_req ]
+    keyUsage = critical,digitalSignature,keyEncipherment
+    extendedKeyUsage = clientAuth
+    ~~~
+
+3. Generate the CA key using the [`openssl genrsa`](https://wiki.openssl.org/index.php/Manual:Genrsa(1)) command:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    openssl genrsa -out ${my-safe-directory}/ca.key 2048 2> /dev/null
+    chmod 400 ${my-safe-directory}/ca.key
+    ~~~
+
+4. Generate the CA certificate using the [`openssl req`](https://wiki.openssl.org/index.php/Manual:Req(1)) command:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    # Create CA certificate.
+    openssl req \
+      -new \
+      -x509 \
+      -config ca.conf \
+      -key ${CCERTS}/ca.key \
+      -out ${OSCERTS}/ca.crt \
+      -days 3660 \
```

Comment:
We should point out the reasoning behind some flags.
Here, just `-days 3660` to match the `cockroach cert` default values.

Reply:
@mberhault I was actually wondering about this. Because we are no longer trying to keep the openssl doc closely parallel to the cert commands doc, can we leave out this parameter? The users can set it to the value they want, right?

Reply:
"Recommended days for expiration of CA certificate"